cmd/tailscale/cli/ssh.go

```diff
@@ -171,38 +217,70 @@ func genKnownHosts(st *ipnstate.Status) []byte {
 			if strings.ContainsAny(hostKey, "\n\r") { // invalid
 				continue
 			}
-			fmt.Fprintf(&buf, "%s %s\n", ps.DNSName, hostKey)
+			// Join all ps.TailscaleIPs as strings separated by commas.
+			ips := make([]string, len(ps.TailscaleIPs))
+			for i, ip := range ps.TailscaleIPs {
+				ips[i] = ip.String()
+			}
+			// Generate comma separated string of all possible names for the host.
+			n := strings.Join(append(ips, ps.DNSName, strings.TrimSuffix(ps.DNSName, "."), strings.Split(ps.DNSName, ".")[0]), ",")
+			fmt.Fprintf(&buf, "%s %s\n", n, hostKey)
```

Comment:
what's this part of the change about? Is this related? Can this be sent out separately in its own PR?

Reply:
The known_hosts file format allows for a comma separated list of hostnames or patterns then a space then the key https://man.openbsd.org/sshd#SSH_KNOWN_HOSTS_FILE_FORMAT. This isn't necessary with your ssh wrapper because you take care of that yourself. But since people can ssh to a device using any of hostname, fqdn, or IP when using the match feature every thing has to be accounted for. This creates a host file that has all the possible names that someone could use to connect to a device and is necessary to the rest of the PR.